src/Compilers/CSharp/Test/Semantic/Semantics/PatternSwitchTests.cs

```diff
@@ -1331,5 +1331,75 @@ public static bool IsB(char value)
                 Diagnostic(ErrorCode.ERR_PatternIsSubsumed, "bool b").WithLocation(11, 18)
                 );
         }
+
+        [Fact, WorkItem(273713, "https://devdiv.visualstudio.com/DevDiv/_workitems?id=273713")]
+        public void ExplicitTupleConversion_Crash_1()
+        {
+            var source =
+@"namespace System
+{
+    public struct ValueTuple<T1, T2>
+    {
+        public T1 Item1;
+
+        public ValueTuple(T1 item1, T2 item2)
+        {
+            this.Item1 = item1;
+        }
+    }
+}
+class Program
+{
+    public static void Main(string[] args)
+    {
+        object o = args;
+        switch (o)
+        {
+            case System.ValueTuple<int, int> z1:
+                break;
+            case System.ValueTuple<long, long> z3:
+                break;
+        }
+    }
+}
+";
+            var compilation = CreateCompilationWithMscorlib45(source, options: TestOptions.DebugExe);
+            compilation.VerifyDiagnostics();
+        }
+
+        [Fact, WorkItem(273713, "https://devdiv.visualstudio.com/DevDiv/_workitems?id=273713")]
+        public void ExplicitTupleConversion_Crash_2()
```

Comment:
Technically, no, but it is the same underlying issue (a kind of conversion that wasn't explicitly handled).